provider/ecs/ecs.go

```diff
@@ -22,6 +22,7 @@ import (
 )
 
 var _ provider.Provider = (*Provider)(nil)
+var existingTaskDef = make(map[string]*ecs.TaskDefinition)
```

Comment:
Could you use a cache that support expiration.

We already use github.com/patrickmn/go-cache inside Traefik.

Reply:
I made this requested change in a separate PR. Please review, thanks!